jetty-io/src/main/java/org/eclipse/jetty/io/ArrayRetainableByteBufferPool.java

```diff
@@ -311,4 +358,51 @@ private void evict(boolean direct, long excess)
         }
         return oldestEntry;
     }
+
+    private static class Bucket extends Pool<RetainableByteBuffer>
+    {
+        private final int _capacity;
+
+        Bucket(int capacity, int size)
+        {
+            super(Pool.StrategyType.THREAD_ID, size, true);
+            _capacity = capacity;
+        }
+
+        @Override
+        public String toString()
+        {
+            return String.format("%s{capacity=%d}", super.toString(), _capacity);
+        }
+    }
+
+    /**
+     * A variant of the {@link ArrayRetainableByteBufferPool} that
+     * uses buckets of buffers that increase in size by a power of
+     * 2 (eg 1k, 2k, 4k, 8k, etc.).
+     */
+    public static class ExponentialPool extends ArrayRetainableByteBufferPool
```

Comment:
Not sure we want to make this `public`, as it binds us to maintain it, etc.

Reply:
I want to make it public so we can easily try using it in real deployments.    
But as there appears to be resistance to this idea, I'll move to test and we can used the sampled one to gather data to see if `ExponentialPool` is worthwhile or not.